tools.go

```diff
@@ -0,0 +1,7 @@
+// +build tools
+
+package tools
+
+import (
+	_ "google.golang.org/protobuf/cmd/protoc-gen-go"
```

Comment:
In order to convert .proto to .pb.go, two parts are required:

- `protoc` compiler (which is out of the scope of this change)
- `protoc-gen-go` which serves as a plugin for `protoc` and specifically for golang

The idea of this change is to use the same dependency management (go.mod) of this project to ensure a consistent version of protoc-gen-go to be used.

Reply:
Thanks for explanation. could you tell me how this file work? where is the build tag tools used? I think you want to pin down the version of protoc-gen-go in go.mod by this change so that "go install google.golang.org/protobuf/cmd/protoc-gen-go" is using fixed version. but I don't see this commit changing go.mod. wondering is this file really taking effect

Reply:
> I think you want to pin down the version of protoc-gen-go in go.mod by this change so that "go install google.golang.org/protobuf/cmd/protoc-gen-go" is using fixed version.

That's correct

> but I don't see this commit changing go.mod.

Because it's pinned already https://github.com/google/seesaw/blob/master/go.mod#L13

> wondering is this file really taking effect

To validate that, change the version from the above line in go.mod (say from v1.23.0 to v1.24.0), run `make proto` with this PR, then the version signature in the below will be changed. Validated that before submitting the change

https://github.com/google/seesaw/blob/master/pb/config/config.pb.go#L21

> could you tell me how this file work? where is the build tag tools used?

This is because `go install` is module aware.

Reply:
> Because it's pinned already https://github.com/google/seesaw/blob/master/go.mod#L13

yeah. that's basically where my confusion comes from. This is file is not necessary because go.mod is already pinning the deps. But it doesn't hurt to have this file.

Reply:
I see. So currently (before this PR), the reason for that dependency is because of the generated pb.go.

> $  go mod why -m  google.golang.org/protobuf
> # google.golang.org/protobuf
> github.com/google/seesaw/pb/config
> google.golang.org/protobuf/reflect/protoreflect

But actually it'd better to call it out explicitly, that `google.golang.org/protobuf/cmd/protoc-gen-go` is a dependency of seesaw project itself. That's the purpose of having this tools.go

And the build tags never supposed to be used, as the sole purpose is to declare the dependency through the file for golang toolchain to be aware of.